module/VuFind/tests/unit-tests/src/VuFindTest/Auth/LoginTokenTest.php

```diff
@@ -0,0 +1,249 @@
+            ->disableOriginalConstructor()
+            ->getMock();
+        return $table;
+    }
+
+    /**
+     * Get a mock user.
+     *
+     * @return User
+     */
+    protected function getMockUser()
+    {
+        $userData = [
+            ['id', 0]
+        ];
+        $user = $this->getMockBuilder(\VuFind\Db\Row\User::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+        $user->method('__get')
+            ->will($this->returnValueMap($userData));
+        $user->method('offsetGet')
+            ->will($this->returnValueMap($userData));
+        return $user;
+    }
+
+    /**
+     * Get a mock Login Token.
+     *
+     * @return User
+     */
+    protected function getMockLoginToken()
+    {
+        $tokenData = [
+            ['token', '111'],
+            ['user_id', 0],
+            ['series', '222'],
+            ['expires', 2]
+        ];
+        $token = $this->getMockBuilder(\VuFind\Db\Row\LoginToken::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+        $token->method('__get')
+            ->will($this->returnValueMap($tokenData));
+        $token->method('offsetGet')
+            ->will($this->returnValueMap($tokenData));
+        return $token;
+    }
+
+    /**
+     * Get a mock user table.
+     *
+     * @return LoginTokenTable
+     */
+    protected function getMockLoginTokenTable()
+    {
+        $table = $this->getMockBuilder(\VuFind\Db\Table\LoginToken::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+        return $table;
+    }
+
+    /**
+     * Get cookie manager
+     *
+     * @param array $cookies Cookies
+     *
+     * @return CookieManager
+     */
+    protected function getCookieManager(array $cookies): CookieManager
+    {
+        return new CookieManager(
+            $cookies,
+            '/first',
+            'localhost',
+            false,
+            'SESS',
+        );
+    }
+
+    /**
+     * Get login token
+     *
+     * @param CookieManager $cookieManager cookie manager
+     * @param LoginToken    $tokenTable    Login token table
+     * @param USer          $userTable     User table
+     *
+     * @return LoginToken
+     */
+    protected function getLoginToken($cookieManager, $tokenTable, $userTable)
+    {
+        $config = new Config([]);
+        $sessionManager = new SessionManager();
+        $mailer = $this->getMockBuilder(\VuFind\Mailer\Mailer::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+        $viewRenderer = $this->getMockBuilder(\Laminas\View\Renderer\RendererInterface::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+        return new LoginToken(
+            $config,
+            $userTable,
+            $tokenTable,
+            $cookieManager,
+            $sessionManager,
+            $mailer,
+            $viewRenderer
+        );
+    }
+}
```

Comment:
This file is missing a newline at the end -- one of a few minor style issues that you can probably fix automatically with `vendor/bin/phing phpcbf`.